src/components/node-list/node-list-row.js

```diff
@@ -76,29 +77,32 @@ const NodeListRow = memo(
         title={name}
         onMouseEnter={visible ? onMouseEnter : null}
         onMouseLeave={visible ? onMouseLeave : null}>
+        {type && (
+          <NodeIcon
+            className={classnames(
+              'pipeline-nodelist__row__type-icon',
+              'pipeline-nodelist__row__icon',
+              {
+                'pipeline-nodelist__row__type-icon--faded': faded,
+                'pipeline-nodelist__row__type-icon--disabled': disabled,
+                'pipeline-nodelist__row__type-icon--nested': !children,
+                'pipeline-nodelist__row__type-icon--active': active,
+                'pipeline-nodelist__row__type-icon--selected': selected,
+              }
+            )}
+            type={type}
+          />
+        )}
         <TextButton
-          className="pipeline-nodelist__row__text"
+          className={classnames(
+            'pipeline-nodelist__row__text',
+            `pipeline-nodelist__row__text--kind-${kind}`
```

Comment:
for readability purposes I would format this as `{'pipeline-nodelist__row__text--kind-filter': kind === 'filter'}` , given that there aren't any classes applied other than for the `filter` type rows.

Reply:
I think we've discussed but this is just a pattern to allow styling of multiple kinds, I think it's worth keeping as a pattern as the sidebar evolves there may be more.